gcc compilation errors

```
  In file included from c/python/obj.h:13:0,
                   from c/types.cc:8:
  c/types.h: In function ‘bool ISNA(T) [with T = float]’:
  c/types.h:390:57: error: ‘isnan’ was not declared in this scope
   template<> inline bool ISNA(float x)    { return isnan(x); }
                                                           ^
  c/types.h:390:57: note: suggested alternative:
  In file included from c/types.h:11:0,
                   from c/python/obj.h:13,
                   from c/types.cc:8:
  /usr/include/c++/5/cmath:641:5: note:   ‘std::isnan’
       isnan(_Tp __x)
       ^
  In file included from c/python/obj.h:13:0,
                   from c/types.cc:8:
  c/types.h: In function ‘bool ISNA(T) [with T = double]’:
  c/types.h:391:57: error: ‘isnan’ was not declared in this scope
   template<> inline bool ISNA(double x)   { return isnan(x); }
                                                           ^
  c/types.h:391:57: note: suggested alternative:
  In file included from c/types.h:11:0,
                   from c/python/obj.h:13,
                   from c/types.cc:8:
  /usr/include/c++/5/cmath:641:5: note:   ‘std::isnan’
       isnan(_Tp __x)
       ^
  c/types.cc: In member function ‘const char* info::ltype_name() const’:
  c/types.cc:328:1: warning: control reaches end of non-void function [-Wreturn-type]
   }
   ^
  error: command 'gcc' failed with exit status 1
```
with `gcc (Ubuntu/IBM 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609`

--- a/c/csv/toa.h
+++ b/c/csv/toa.h
@@ -44,7 +44,7 @@ inline void htoa(char** pch, int16_t value)
     *ch++ = '-';
     value = -value;
   }
-  int r = (value < 1000)? 2 : 4;
+  unsigned int r = (value < 1000)? 2 : 4;
   for (; value < DIVS32[r]; r--);
   for (; r; r--) {
     int d = value / DIVS32[r];

--- a/c/extras/py_validator.h
+++ b/c/extras/py_validator.h
@@ -50,7 +50,7 @@ void check_positive(T value,
                     const std::string& name = _name,
                     error_manager& em = _em)
 {
-  if (!isinf(value) && value > 0) return;
+  if (!std::isinf(value) && value > 0) return;
   throw em.error_not_positive(o.to_borrowed_ref(), name);
 }
 
@@ -65,7 +65,7 @@ void check_not_negative(T value,
                         const std::string& name = _name,
                         error_manager& em = _em)
 {
-  if (!isinf(value) && value >= 0) return;
+  if (!std::isinf(value) && value >= 0) return;
   throw em.error_negative(o.to_borrowed_ref(), name);
 }
 

--- a/c/mmm.h
+++ b/c/mmm.h
@@ -8,6 +8,7 @@
 #ifndef dt_MMM_h
 #define dt_MMM_h
 #include <vector>
+using std::size_t;
 
 
 class MemoryMapWorker {

--- a/c/rowindex_array.cc
+++ b/c/rowindex_array.cc
@@ -30,13 +30,15 @@
 #include "utils/assert.h"
 #include "utils/parallel.h"
 
-static void test(ArrayRowIndexImpl* o) {
-  #ifndef NDEBUG
+#ifndef NDEBUG
+  inline static void test(ArrayRowIndexImpl* o) {
     o->refcount++;
     o->verify_integrity();
     o->refcount--;
-  #endif
-}
+  }
+#else
+  inline static void test(ArrayRowIndexImpl*) {}
+#endif
 
 
 

--- a/c/types.cc
+++ b/c/types.cc
@@ -315,7 +315,8 @@ LType info::ltype() const {
 }
 
 const char* info::ltype_name() const {
-  switch (ltype()) {
+  LType lt = ltype();
+  switch (lt) {
     case LType::MU:       return "void";
     case LType::BOOL:     return "bool";
     case LType::INT:      return "int";
@@ -325,6 +326,7 @@ const char* info::ltype_name() const {
     case LType::DURATION: return "duration";
     case LType::OBJECT:   return "obj";
   }
+  throw RuntimeError() << "Unknown ltype " << int(lt); // LCOV_EXCL_LINE
 }
 
 py::oobj info::py_ltype() const {

--- a/c/types.h
+++ b/c/types.h
@@ -387,8 +387,8 @@ template<> inline bool ISNA(int32_t x)  { return x == NA_I4; }
 template<> inline bool ISNA(int64_t x)  { return x == NA_I8; }
 template<> inline bool ISNA(uint32_t x) { return (x & NA_S4); }
 template<> inline bool ISNA(uint64_t x) { return (x & NA_S8); }
-template<> inline bool ISNA(float x)    { return isnan(x); }
-template<> inline bool ISNA(double x)   { return isnan(x); }
+template<> inline bool ISNA(float x)    { return std::isnan(x); }
+template<> inline bool ISNA(double x)   { return std::isnan(x); }
 template<> inline bool ISNA(PyObject* x) { return x == Py_None; }
 
 /**

--- a/datatable/xls.py
+++ b/datatable/xls.py
@@ -259,7 +259,7 @@ def _excel_coords_to_range2d(ec):
             i = i * 26 + (ord(c) - ord('A')) + 1
         return i - 1
 
-    mm = re.match("([A-Z]+)(\d+):([A-Z]+)(\d+)", ec)
+    mm = re.match(r"([A-Z]+)(\d+):([A-Z]+)(\d+)", ec)
     if not mm:
         return None
     row0 = int(mm.group(2)) - 1

